### DISCUSSION TITLE
Allow arbitrary JS on pages

### DISCUSSION THREAD

**Discussion by skamensky** [cortezaproject/corteza-server#808] [Ideas]
A lot of ideas and discussions are around the UI and potential improvements to it.

I think it could be a good idea to allow users to add an "include" to a js source file (or a raw source string) as an escape hatch for advanced users.

For example, if I want to add a hotkey so when users press ctrl+s it saves the page, I don't have to wait for corteza to add it (if they decide to add it), and I also don't need to clone the whole front-end and take it on as a maintenance burden for that small change.

**Comment by tutorempire** [comment]
I would like this as well. A little javascript + jQuery would go a long way towards adding quality-of-life UI improvements for users. 

For example, I'd like to change how URLs are displayed (I don't want to see 20 rows starting with "https://", but there's no way to do it. Or display links to social media (i.e. IG) in a specific way - @username instead of https://instagram.com/username

Lots of other use cases out there.
